controllers/datadogagent/feature/logcollection/feature.go

```diff
@@ -0,0 +1,200 @@
+// It should do nothing if the feature doesn't need to configure it.
+func (f *logCollectionFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	// pointerdir volume mount
+	var pointerVol, logPodVol, logContainerVol, symlinkVol corev1.Volume
+	var pointerVolMount, logPodVolMount, logContainerVolMount, symlinkVolMount corev1.VolumeMount
+	// TODO: mount as ReadWrite instead of ReadOnly
+	if f.tempStoragePath != "" {
+		pointerVol, pointerVolMount = volume.GetVolumes(apicommon.PointerVolumeName, f.tempStoragePath, apicommon.PointerVolumePath, false)
+	} else {
+		pointerVol, pointerVolMount = volume.GetVolumes(apicommon.PointerVolumeName, apicommon.LogTempStoragePath, apicommon.PointerVolumePath, false)
+	}
+	managers.Volume().AddVolumeToContainer(&pointerVol, &pointerVolMount, apicommonv1.CoreAgentContainerName)
+
+	// pod logs volume mount
+	if f.podLogsPath != "" {
+		logPodVol, logPodVolMount = volume.GetVolumes(apicommon.LogPodVolumeName, f.podLogsPath, f.podLogsPath, true)
+	} else {
+		logPodVol, logPodVolMount = volume.GetVolumes(apicommon.LogPodVolumeName, apicommon.LogPodVolumePath, apicommon.LogPodVolumePath, true)
+	}
+	managers.Volume().AddVolumeToContainer(&logPodVol, &logPodVolMount, apicommonv1.CoreAgentContainerName)
+
+	// container logs volume mount
+	if f.containerLogsPath != "" {
+		logContainerVol, logContainerVolMount = volume.GetVolumes(apicommon.LogContainerVolumeName, f.containerLogsPath, f.containerLogsPath, true)
+	} else {
+		logContainerVol, logContainerVolMount = volume.GetVolumes(apicommon.LogContainerVolumeName, apicommon.LogContainerVolumePath, apicommon.LogContainerVolumePath, true)
+	}
+	managers.Volume().AddVolumeToContainer(&logContainerVol, &logContainerVolMount, apicommonv1.CoreAgentContainerName)
+
+	// symlink volume mount
+	if f.containerSymlinksPath != "" {
+		symlinkVol, symlinkVolMount = volume.GetVolumes(apicommon.SymlinkContainerVolumeName, f.containerSymlinksPath, f.containerSymlinksPath, true)
+	} else {
+		symlinkVol, symlinkVolMount = volume.GetVolumes(apicommon.SymlinkContainerVolumeName, apicommon.SymlinkContainerVolumePath, apicommon.SymlinkContainerVolumePath, true)
+	}
+	managers.Volume().AddVolumeToContainer(&symlinkVol, &symlinkVolMount, apicommonv1.CoreAgentContainerName)
+
+	// envvars
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDLogsEnabled,
+		Value: strconv.FormatBool(f.enable),
+	})
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDLogsConfigContainerCollectAll,
+		Value: strconv.FormatBool(f.containerCollectAll),
+	})
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDLogsContainerCollectUsingFiles,
+		Value: strconv.FormatBool(f.containerCollectUsingFiles),
+	})
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
```

Comment:
perhaps just add an `if` conditional here since it's possible `f.openFilesLimit` isn't set